### PR TITLE
doc/dev: add irc registration instructions

### PR DESCRIPTION
**Pull request by zdover23**
I added irc registration instructions to essentials.rst.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/developer_guide/essentials.rst [hunk 1]
```diff
@@ -159,6 +159,26 @@ register an account. Registering gives you a unique IRC identity and allows you
 to access channels where unregistered users have been locked out for technical
 reasons.
 
+To register your account on Ceph's IRC channels, type the following into the text field in your IRC client::
+
+  /msg NickServ REGISTER irc_username username@foo.com
```
Comment (jdurgin) [inline doc/dev/developer_guide/essentials.rst:164]: irc_username here should be password

Comment (zdover23) [reply doc/dev/developer_guide/essentials.rst:164]: I'm not sure about that, Josh. I just checked my irc history, and I ran the command as I have it, not as you're suggesting. It worked.

I know that kind of leaves us at an impasse, and I don't mean to shut the conversation down, so: what now?

Comment (jdurgin) [reply doc/dev/developer_guide/essentials.rst:164]: linking to the oftc docs sounds good, and future-proof

**Comment by neha-ojha** [comment]
@zdover23 ping

**Comment by sebastian-philipp** [comment]
@zdover23 ping

**Comment by ljflores** [comment]
@zdover23 this is definitely still useful.

**Review by anthonyeleven — CHANGES_REQUESTED** [review]
In other lines, could we update these to reflect the current state and domain?


`The Ceph project is led by Sage Weil.`

and

`other Ceph-related mailing lists`: https://ceph.com/irc/

**Comment by zdover23** [comment]
Superseded by https://github.com/ceph/ceph/pull/46938

**Comment by zdover23** [comment]
> 

As of https://github.com/ceph/ceph/pull/46832 (https://github.com/ceph/ceph/pull/46832/commits/83b77909e5c1149fafd03986098bd0256e07b479), Sage's now-historical position is referred to correctly.

**Pull request closed without merging**